### PR TITLE
Rearrange docs

### PR DESCRIPTION
**Pull request by blp**
Is this a user-visible change (yes/no): no

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# Concepts
+
+Feldera Platform processes queries and produces output continuously.  When input
+arrives, Feldera Platform recomputes query results and sends the changes to
+outputs.  Feldera Platform queries are written in SQL, so users who have an
+existing investment in analyzing data at rest with a SQL database can
+use much the same code to analyze data continuously with Feldera Platform.
```
Comment (lalithsuresh) [inline docs/concepts.md:7]: use much of the same

File: docs/concepts.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# Concepts
+
+Feldera Platform processes queries and produces output continuously.  When input
```
Comment (lalithsuresh) [inline docs/concepts.md:3]: Not sure if we should continue to stick to "Feldera Platform". Maybe "Feldera Engine" or "Feldera" in short? cc @ryzhyk

File: docs/concepts.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+use much the same code to analyze data continuously with Feldera Platform.
+
+## Incremental processing
+
+Feldera Platform is fundamentally **incremental** in how it handles input,
+computation, and output.
+
+For input, being incremental means that Feldera Platform processes data as it
+arrives. Feldera Platform does not require all of the data to
+be on hand before beginning computation. Unlike a database, Feldera Platform does
```
Comment (lalithsuresh) [inline docs/concepts.md:16]: We do durably store data. :) What we don't do is ad-hoc queries.

File: docs/concepts.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+not durably store data.
+
+For computation, being incremental means that when new data arrives,
+Feldera Platform does a minimal amount of work to update query results, rather
+than by fully recomputing them. This speeds up processing.
```
Comment (lalithsuresh) [inline docs/concepts.md:21]: than by fully recomputing them from scratch.

File: docs/concepts.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+For output, being incremental means that Feldera Platform outputs query results as
+sets of changes from previous output. This improves performance and
```
Comment (lalithsuresh) [inline docs/concepts.md:24]: from the previous output.

**Review by lalithsuresh — COMMENTED** [review]
It's only after I read through for a while that I realized this PR was mainly moving the files around. We'd need a pretty big revision of the content, so ignore my specific comments. :)

**Comment by blp** [comment]
> It's only after I read through for a while that I realized this PR was mainly moving the files around. We'd need a pretty big revision of the content, so ignore my specific comments. :)

Oh, yeah, I was just getting started by getting the overall structure correct.

**Pull request merged**